system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -3226,20 +3226,22 @@ class CLdapClient : implements ILdapClient, public CInterface
         return true;
     }
 
-    virtual bool changePasswordSSL(const char* username, const char* newPassword)
+    virtual bool changePasswordSSL(const char* username, const char* newPassword, LDAP* ld)
     {
         Owned<ILdapConnection> lconn;
-        try
-        {
-            lconn.setown(m_connections->getSSLConnection());
-        }
-        catch(IException*)
+        if (ld == nullptr)
         {
-            throw MakeStringException(-1, "Failed to set user %s's password because of not being able to create an SSL connection to the ldap server. To set an Active Directory user's password from Linux, you need to enable SSL on the Active Directory ldap server", username);
+            try
+            {
+                lconn.setown(m_connections->getSSLConnection());
+            }
+            catch (IException *)
```

Comment:
I know this is only a diff because of whitespace, but this looks like a leak. The exception should be named so it can be released before throwing a different exception in its place.